nixos/modules/flyingcircus/roles/mongodb/default.nix

```diff
@@ -144,13 +144,11 @@ in
       critical = 63000;
     };
 
-    services.telegraf.extraConfig = {
-      inputs = {
-        mongodb = {
-          servers = ["mongodb://127.0.0.1:27017"];
-          gather_perdb_stats = true;
-        };
-      };
+    services.telegraf.inputs = {
+      mongodb = [{
+        servers = ["mongodb://127.0.0.1:27017"];
```

Comment:
Nix standard list style is: `[ elem elem elem ]`. Note the spaces inside brackets.